doc/command_ref.rst

```diff
@@ -1746,11 +1747,12 @@ Specifying Packages
 
 Many commands take a ``<package-spec>`` parameter that selects a package for
 the operation. The ``<package-spec>`` argument is matched against package
-NEVRAs, provides and file provides.
+:ref:`NEVRAs <specifying_nevra-label>`, :ref:`provides <specifying_provides-label>` and
+:ref:`file provides <specifying_file-provides-label>`.
```

Comment:
Your changes don't fit well with the structure of the sections, it's like this:
```
.. _specifying_packages-label:

===================
Specifying Packages
===================
Talking about `<package-spec>` and `<package-file-spec>`.

.. _specifying_provides-label:

===================
Specifying Provides
===================
Talking about `<provide-spec>`.

.. _specifying_file-provides-label:

========================
Specifying File Provides
========================
Your new section that seems to just be a note that belongs under "Specifying Packages", unless we have `<file-provide-spec>` in some command?
```

Maybe you should add it as:
```
------------------------
Specifying File Provides
------------------------
```

Under the `Specifying Packages` section?

And subsequently the refs you're adding on the above lines go across these sections and IMO shouldn't be there. Maybe this could use some re-structuring, but (or because? :slightly_smiling_face:) it's quite complicated...

Reply:
@lukash I changed it. Hope that it is better.

Reply:
I suggest that it is correct to have it in `Specifying Provides` because file provides are mentioned there as a subtype.

Reply:
Well, file provides are also relevant and discussed in `Specifying Packages`, I guess it should go into both, which is the real problem here, there's a mixture of stuff and most, but not all, combinations are relevant and related to each other.

Could you please remove the added refs on lines above (1750 and 1751)? They're not clean in the sense that one points to a sub-section in this section (nevra-label), one points to another section (provides-label) and your new one points to a subsection of another section. I think they're not necessary. Ideally we'd restructure this (and make things yet more complex) but let's just keep it like it is? I think your short new note doesn't need a label...

Reply:
I am sorry, but I suggest that references improve it. The most of the people will not look what is file provide because it is clear, therefore they will not know how to use them with globs. When we will we keep the reference they might think that there is something important and they will look at. Also it was the major point of related bugzilla that the detection of file provides is strange.

Reply:
Yes, but they go across sections and are not consistent. For example the `specifying_provides-label` points to the description of `<provide-spec>`, which is just a different thing than `<package-spec>`. If you want to keep the references, then please restructure the whole thing so that the references are consistently pointing to the right things...